lib/adapter.js

```diff
@@ -37,74 +41,83 @@ class CucumberAdapter {
         this.capabilities = capabilities
 
         this.cucumberOpts = Object.assign(DEFAULT_OPTS, config.cucumberOpts)
```

Comment:
Thanks for the tip @pantherqin and for the reminder that this is not merged yet! I'm about to try your suggestion and let you know how this goes.

as for CLI, per this comment:
https://github.com/cucumber/cucumber-js/issues/786#issuecomment-286793230

and also, I did start with CLI at first but I encountered some limitations that seemed almost impossible to solve (I can't recall right now, it was hundreds if not thousands of lines of code ago ;) ). Maybe there wasn't a way to attach custom listeners?